### ISSUE TITLE
bug: when type single or double quote, it automatically output double single quotes or double double quotes

### ISSUE THREAD

**Issue by dlintw** [LazyVim/LazyVim#1364]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.8.0

### Operating system/version

Rocky Linux 8.6

### Describe the bug

I'm edit a bash script or txt file, it is very annoy when I  type in single quote, it auto type 2 single quotes.
And the double quote have the same behavior.

I don't know how to disable this function,  my suggestion for lazyvim, it is better disable this annoy thing by default.

### Steps To Reproduce

1. nvim foo.txt
2. type single quote
3. it shows 2 single quote

### Expected Behavior

1. nvim foo.txt
2. type single quote
3. it shows 1 single quote

### Repro

```Lua
default lazyvim setup.
```

**Comment by dlintw** [comment]
seems possible disable by add this line into the lazy.lua
```
 { "lazy/mini.pairs", enabled = false },
```

**Comment by dlintw** [comment]
add this in <configpath>/lua/config/options.lua will disable this function. maybe setup a <leader> hotkey to toggle it will be better
```
vim.g.minipairs_disable = true
```

**Comment by abeldekat** [comment]
> I don't know how to disable this function, my suggestion for lazyvim, it is better disable this annoy thing by default.

@dlintw,

When you create an issue, you agree with the following:

> I have read all the LazyVim docs *

Obviously, you did not. Instead of making an effort to familiarize yourself with [LazyVim](https://www.lazyvim.org/), you invested the time to write this issue, even qualifying a core plugin as *this annoy thing*.

**Comment by dpetka2001** [comment]
@dlintw As @abeldekat mentioned, it seems like you didn't go through other issues to see if your problem is mentioned somewhere else. Please take a look at #1357 (it also mentions how to bypass this issue if you don't want to remove `mini.pairs`). It seems to me like this is an entirely personal preference, so of course you can make that change to your personal configuration as you see fit. Unfortunately, not all defaults will cater to everyone's needs and that's why we tinker with things in our personal configuration, to add or remove certain features that don't meet our workflow. Since this is not a Lazyvim bug, I would suggest maybe you close this issue and if you want further help move to the Discussions section.

**Comment by abeldekat** [comment]
Hi @dpetka2001,

Thanks for your kind rewording. However, this is not entirely what I mentioned. In a previous issue, @dlintw  was **already** given information about the discussion's forum, some days ago. He did not reply.

I think this line is plain disrespectful:

> I don't know how to disable this function, my suggestion for lazyvim, it is better disable this annoy thing by default.

**Comment by dpetka2001** [comment]
@abeldekat Oh I'm sorry I wasn't aware of the previous issue or maybe just slipped my mind. My bad, didn't have any intention of miswording what you said. I agree with the disrespectful point of yours totally. Plz excuse my slip up. My point was more focused on how Lazyvim's defaults cannot cater to everyone's needs, thus why we change stuff ourselves in our personal configs.

**Comment by abeldekat** [comment]
@dpetka2001,

Your comment is, as always, very well written in my opinion. I agree completely, there is no need to apologize!

**Comment by dlintw** [comment]
This annoy default should disable or show an example disable in options.lua.  As a newbie it is too hard to figure out which hotkey/plugin cause this behaviour.

I like the lazyvim's hotkey document, it almost summarize the important key feature of nvim.

What I treat lazyvim is a good entry to enter lua config nvim.  In general, all default suitable for a newbie.

**Comment by dpetka2001** [comment]
@dlintw You should understand that the annoying default you're referring to, is just a personal preference. There is already another issue open with similar context to your problem #1357, where it has been discussed how to get rid of this issue for people who don't like it. You can either disable `mini.pairs` to get rid of this functionality as a whole or you can press `C-v` while in `insert` mode and then you will be able to insert a single character without it being interpreted as a pair by `mini.pairs`. This is not by any means to be meant as a discouragement towards you, rather a suggestion that you should look more carefully for other issues that already may involve the problem that you're facing. Since this is not a Lazyvim bug, I would once more suggest that you close this issue and ask for further help in the Discussions if you're in such need. Plz bear in mind I'm merely making a suggestion and you're under no obligation to do that if you think otherwise. It's just that, in my personal opinion. issues should be made for actual bugs in Lazyvim and any help for personal configuration should be made in the Discussions instead.

**Comment by traverseda** [comment]
For what it's worth, I searched for how to disable this in a few different ways, and this was the only ticket that came up. It's a bit more descriptive in the terms it uses if you don't know that you should be searching for "mini.pairs". If OP hadn't made this ticket someone else would have, if only because it's very difficult to search for on it's own.

This is also very annoying for quotations under python, where triple qoutes `"""` are reasonably common, and where it has some automatic behavior where it will delete multiple quotes if you start deleting from the wrong place, leads to me having too many quotes, than too few, than too many. Might be worth seriously reconsidering this feature, or maybe someone should just open a bug with upstream. I'm happy with it just being disabled, but I think this probably is a point of frisson for many of your users.

**Comment by dpetka2001** [comment]
Neovim is all about customization.

 Everyone tinkers their own personal configuration to cater to their needs. Unfortunately, not everyone's needs are able to align with each other and that should be perfectly fine. 

I'm just a user like you and I have disabled `mini.pairs` in favor of `nvim-autopairs`, which also seems to handle better the triple quotes in Python as far as I can tell from my limited Python experience. 

If you believe that something should be considered a bug for a plugin, then feel free to create an issue upstream at the corresponding plugin's repo, because LazyVim is just a distro that makes use of plugins available in the Neovim ecosystem.

**Comment by traverseda** [comment]
Understood, I've been using vim for a long time. You need to understand that you are setting the defaults for your users. Is their a process for deciding is `mini.pairs` is a reasonable default compared to other plugins with equivalent functionality?

I'm not going to champion this issue, I've moved to lazyvim because my free time is rapidly dwindling. But I will say that this default is not ergonomic, and personally I don't think it should be a default how it's configured now. Not my project, not something I'm going to try to actually make happen, just adding another voice saying "Hey, this doesn't actually work very well, why is it included in the easy vim distro?".

**Comment by dpetka2001** [comment]
Defaults are just personal bias. Every distro is in some way or another affected by their creator's personal preferences.

It's not my place to judge other people's personal preferences. I don't have a say why the maintainer made this choice. I do, however, have the choice to make any changes I deem necessary in my personal configuration (that's why i ended up disabling this myself as well).

As I said, Neovim is about customization and not everyone's needs will align. Just because you think this is not an ergonomic default, doesn't mean other people share the same opinion.

**Comment by traverseda** [comment]
Yes, we're all aware that there's no objective criteria for best vim distro. That doesn't mean subjective criteria, like "this doesn't work for me in these situations" matters any less though. Good defaults do matter, and *subjectively* I don't think this is a good default. Apparently neither do you since you got rid of it.

Does anyone think this is an ergonomic default? Would any new users complain if this plugin was swapped out for one that didn't mangle triple quotes? I don't think so, and  while ultimately that is a subjectivie decision made by a human being, I can provide another voice saying "hey, this doesn't work for me, can we improve the defaults". That doesn't make it any less of a subjective decision, but I'm aware of that, you're aware of that, the people making the decision is aware of that. You're kind of just saying "well there's no such thing as objective truth so nothing matters", which sure, but there's still subjective criteria you can use to try to make these kinds of decisions.

Yes, I get your point, things are subjective, but ultimately everything is subjective, so the fact that this technical decision is subjective just means it's a decisions, not that it's a special type of decision that we can't discuss the merits of.

**Comment by traverseda** [comment]
https://blog.codinghorror.com/the-power-of-defaults/

**Comment by dpetka2001** [comment]
My opinion can be relatively simply summarized in that Neovim is an editor that I can customize to my needs. I never saw any Neovim distro as something that uses good defaults. The defaults I can change to whatever I want. What draws me to a Neovim distro (personal opinion always) is the ease of being able to customize things to my needs.

You're right that I removed the plugin because it didn't suit my needs. However, I'm not criticizing it on a `good defaults` basis. I don't expect from Neovim itself and every Neovim distro that exists to suit what I consider "good defaults". I would've used vscode for that.

I use Neovim, because I like to have control over what I can customize. Thus, I really don't care about ergonomics like these where one plugin does something better than another. I can just switch them. 

The ergonomics I care about are more in the scope of better LSP servers implementation that can offer better experience compared to IntelliJ and other products. 

In the end I just prefer to not judge people's personal preferences. For me, I see it more as "Ok the maintainer liked this for some reason, I don't have to like it as well, but it's also not a big deal to have it as default since I can just change it and do whatever I want" kind of thing. There's already in the Discussions section a thread about swapping `mini.pairs` with `nvim-autopairs`, which I really like and hope maybe the maintainer will have a look at it when he gets back. But even if nothing comes out of it, so what? I will still do what I want in my personal configuration.

**Comment by traverseda** [comment]
Sure, care to post a link to that discussion form page? That's very relevant information for people who might look at this issue in the future.

**Comment by dpetka2001** [comment]
Sure, [here](https://github.com/LazyVim/LazyVim/discussions/2248) you go.
